data/transactions/logic/assembler.go

```diff
@@ -2680,6 +2680,10 @@ func disassemble(dis *disassembleState, spec *OpSpec) (string, error) {
 
 			pc++
 		case immLabel:
+			// decodeBranchOffset assumes it has two bytes to work with
+			if pc+2 > len(dis.program) {
```

Comment:
my understanding is that, `decodeBranchOffset` reads `pc` and `pc + 1`, and this condition blocks index out of scope in `decodeBranchOffset`, the index after offset can at most be `len(dis.program)`.

Reply:
Since decodeBranchOffset reads `pc+1`, we're in trouble if `pc+1` is greater than _or equal_ to the length. That's the same as saying we're in trouble if `pc+2` if _greater_ than the length.

I wrote it that way so that the number "2" would appear, since that's the number of bytes after the opcode we're reading, and I wanted it to read similar to the test above for single byte immediates. Both add the number of bytes they intend to read after the opcode to `pc` and compare greater to len.

Reply:
makes sense, resolving